src/cr/cube/dimension.py

```diff
@@ -19,124 +21,125 @@ class Dimension(object):
     :attr:`.CrunchCube.dimensions`.
     """
 
-    def __init__(self, dim, selections=None):
-        self._dim = dim
-        self._type = self._get_type(dim, selections)
+    def __init__(self, dimension_dict, next_dimension_dict=None):
+        self._dimension_dict = dimension_dict
+        self._next_dimension_dict = next_dimension_dict
```

Comment:
I renamed `dim` to the more descriptive `_dimension_dict`, especially since `dim` is so often used as a shorthand for a `Dimension` object. I renamed `selections` to the more precise `next_dimension_dict` for clarity.